x/btcstaking/keeper/btc_delegations.go

```diff
@@ -57,11 +57,28 @@ func (k Keeper) AddBTCDelegation(ctx sdk.Context, btcDel *types.BTCDelegation) e
 		panic(fmt.Errorf("failed to emit EventBTCDelegationStateUpdate for the new pending BTC delegation: %w", err))
 	}
 
+	if err := ctx.EventManager().EmitTypedEvent(types.NewBtcDelCreationEvent(
```

Comment:
Can L51-58 be removed?

Reply:
hmm we should emit both events, but it can be simplified by using ` EmitTypedEvents(tevs ...proto.Message)` . Will do that.

Reply:
Why is it the case? in my mind the former event only contains a subset of the latter event and they share the same type of consumers

Reply:
there are at least two advantages:
1.  consumers that do not want to process the "fat" event can only follow the state state updates events
2. consumers know the initial state of created delegation i.e that is is `PENDING`. Without it they would need to either assume some state or we would need to add state field to `EventBTCDelegationCreated`

Reply:
Hmm I'm not sure if there is a use case for the "slim" event, as the main consumer of this API is backend and other components do not seem to be interested in a newly pending delegation. Also `EventBTCDelegationCreated` seems to imply this BTC delegation is pending, and if not subsequent events will be emitted. That's why I would still prefer to keep each state transition to have a unique type of event.

Another thing is that the `EventBTCDelegationStateUpdate` event now has two use cases, one for voting power update and the other for serving API consumers. This is not a perfect idea as `EventBTCDelegationStateUpdate` is in fact a "DB" object. Now it seems to be a good time to fix this

Reply:
> Another thing is that the EventBTCDelegationStateUpdate event now has two use cases, one for voting power update and the other for serving API consumers. This is not a perfect idea as EventBTCDelegationStateUpdate is in fact a "DB" object. Now it seems to be a good time to fix this

This is good point, we should not expose db objects  for external consumers. Either way, I would not tangle this into this pr which is about new events for api but create separate one for this purpose

Being totally honest I do not have strong preference here and imo both approaches have its merits. @jrwbabylonlab what would api prefer:
1. Some special events, like `delegation_creation`  +  state transition events (i.e one event showing what is current state of delegation)
2. Or set of different events like: `delegation_creation`, `proof_of_inclusion_provided`, `unbonded_early`, `expired` and `covenant_quorum_reached` ?

Reply:
I would avoid the flags as much as possible. 

We could attach new state to every specific event if this would help the api ?